libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs

```diff
@@ -423,7 +424,12 @@ data MLSMessageResponse
   = MLSMessageResponseError GalleyError
   | MLSMessageResponseProtocolError Text
   | MLSMessageResponseProposalFailure Wai.Error
-  | MLSMessageResponseUpdates [ConversationUpdate] (Maybe UnreachableUsers)
+  | -- | The conversation-owning backend could not reach some of the backends that
+    -- have users in the conversation when processing a commit.
+    MLSMessageResponseUnreachableBackends (Set Domain)
+  | -- | If the list of unreachable users is non-empty, it corresponds to users
+    -- that an application message could not be sent to.
+    MLSMessageResponseUpdates [ConversationUpdate] (Maybe UnreachableUsers)
```

Comment:
`Maybe (NonEmpty ..)` instead of just `[..]` is a bit silly, but ok.  :)  that discussion doesn't belong into this PR.

Reply:
Also, can you explain in the haddocks how users can be unreachable, if not by their backend being unreachable?  (If this is obvious to most readers let me know; I have very little context on MLS yet.)

Reply:
> `Maybe (NonEmpty ..)` instead of just `[..]` is a bit silly, but ok. :) that discussion doesn't belong into this PR.

They are isomorphic. I did get some mileage out of my choice over the alternative, but nothing big.

Reply:
> Also, can you explain in the haddocks how users can be unreachable, if not by their backend being unreachable? (If this is obvious to most readers let me know; I have very little context on MLS yet.)

This comes by extension: if a remote backend is unreachable, all of the users on that backend are unreachable too.

Are you confused by the fact I introduced two data constructors for seemingly the same thing ? In the first one, a commit is considered (therefore, an MLS protocol message) and it either fails (e.g., because some backends are unreachable) or succeeds (all backends are reachable); no partiality allowed. In the second one, an application message is considered and it is fine if it cannot be delivered to some users.

If you can clarify this, that would be great as that would help me understand what should be added to haddocks.